pym/bob/input.py

```diff
@@ -2317,24 +2319,29 @@ def prepare(self, inputEnv, sandboxEnabled, inputStates, inputSandbox=None,
             if dep.condition and not all(env.evaluate(cond, "dependency "+recipe)
                                                       for cond in dep.condition): continue
 
-            if dep.toolOverride:
-                try:
-                    thisDepTools = depTools.derive({
-                        k : depTools[v] for k,v in dep.toolOverride.items() })
-                except KeyError as e:
-                    raise ParseError("Cannot remap unkown tool '{}' for dependency '{}'!"
-                        .format(e.args[0], recipe))
-                thisDepDiffTools = depDiffTools.copy()
-                thisDepDiffTools.update({
-                    k : depDiffTools.get(v, v)
-                    for k, v in dep.toolOverride.items() })
+            if dep.clean:
```

Comment:
I would rather arrange it in a way where, even though `inherit` is set to `false`, some parts can still be inherited explicitly. Consider the following example:

```yaml
- name: some::root::recipe
  inherit: false
  environment:
    FOO: "$FOO"
  tools:
    host-toolchain: host-toolchain
```

This would only inherit the `FOO` environment variable and the `host-toolchain` tool. Right now the `environment` and `tools` keys would be completely ignored!

Reply:
Thanks - I changed this and it seam to work now. not sure if I got this `thisDepDiffTools` thing right...

Reply:
I'm afraid it was not fully correct. I've took the freedom to just push an update. Please check if this still works for you...